File: app/src/main/java/com/divercity/android/repository/data/DataRepository.java
package com.divercity.android.repository.data;

import com.divercity.android.data.entity.base.DataArray;
import com.divercity.android.data.entity.company.createcompanybody.CreateCompanyBody;
import com.divercity.android.data.entity.company.response.CompanyResponse;
import com.divercity.android.data.entity.company.sizes.CompanySizeResponse;
import com.divercity.android.data.entity.industry.IndustryResponse;
import com.divercity.android.data.entity.interests.InterestsResponse;
import com.divercity.android.data.entity.location.LocationResponse;
import com.divercity.android.data.entity.major.MajorResponse;
import com.divercity.android.data.entity.occupationofinterests.OOIResponse;
import com.divercity.android.data.entity.school.SchoolResponse;
import com.divercity.android.data.entity.skills.SkillResponse;

import java.util.List;

import io.reactivex.Observable;

/**
 * Created by lucas on 18/10/2018.
 */

public interface DataRepository {

    Observable<List<CompanyResponse>> fetchCompanies(int page, int size, String query);

    Observable<List<IndustryResponse>> fetchIndustries(int page, int size, String query);

    Observable<DataArray<SchoolResponse>> fetchSchool(int page, int size, String query);

    Observable<DataArray<MajorResponse>> fetchMajors(int page, int size, String query);

    Observable<List<LocationResponse>> fetchLocations(int page, int size, String query);

    Observable<List<SkillResponse>> fetchSkills(int page, int size, String query);

    Observable<Boolean> createCompany(CreateCompanyBody body);

    Observable<List<CompanySizeResponse>> fetchCompanySizes();

    Observable<List<InterestsResponse>> fetchInterests();

    Observable<List<OOIResponse>> fetchOccupationOfInterests(
            int pageNumber,
            int size,
            String query);

}

File: app/src/main/java/com/divercity/android/repository/data/DataRepository.kt
package com.divercity.android.repository.data

import com.divercity.android.data.entity.company.createcompanybody.CreateCompanyBody
import com.divercity.android.data.entity.company.response.CompanyResponse
import com.divercity.android.data.entity.company.sizes.CompanySizeResponse
import com.divercity.android.data.entity.industry.IndustryResponse
import com.divercity.android.data.entity.interests.InterestsResponse
import com.divercity.android.data.entity.location.LocationResponse
import com.divercity.android.data.entity.occupationofinterests.OOIResponse
import com.divercity.android.data.entity.photo.PhotoEntityResponse
import com.divercity.android.data.entity.recommendedjobsgoi.RecommendedJobsGOIResponse
import com.divercity.android.data.entity.skills.SkillResponse
import com.divercity.android.model.Degree
import com.divercity.android.model.Major
import com.divercity.android.model.School
import io.reactivex.Observable

/**
 * Created by lucas on 18/10/2018.
 */

interface DataRepository {

    fun fetchCompanies(page: Int, size: Int, query: String?): Observable<List<CompanyResponse>>

    fun fetchIndustries(page: Int, size: Int, query: String?): Observable<List<IndustryResponse>>

    fun fetchSchool(page: Int, size: Int, query: String?): Observable<List<School>>

    fun fetchMajors(page: Int, size: Int, query: String?): Observable<List<Major>>

    fun fetchLocations(page: Int, size: Int, query: String?): Observable<List<LocationResponse>>

    fun fetchSkills(page: Int, size: Int, query: String?): Observable<List<SkillResponse>>

    fun createCompany(body: CreateCompanyBody): Observable<Boolean>

    fun fetchCompanySizes(): Observable<List<CompanySizeResponse>>

    fun fetchInterests(): Observable<List<InterestsResponse>>

    fun searchPhotos(query: String): Observable<List<PhotoEntityResponse>>

    fun fetchOccupationOfInterests(
        pageNumber: Int,
        size: Int,
        query: String?
    ): Observable<List<OOIResponse>>

    fun fetchDegrees(): Observable<List<Degree>>

    fun fetchRecommendedJobsGOIS(
        pageNumber: Int,
        size: Int
    ): Observable<RecommendedJobsGOIResponse>
}
